### ISSUE TITLE
[Bug]: Trail overrides background color / image

### ISSUE THREAD

**Issue by ryanpwaldon** [tsparticles/tsparticles#4882]
### What happened?

When adding a trail via `options.particles.move.trail`, it seems to override whatever background color / image I've set. Is this a limitation of using trail?

Repro: https://codepen.io/ryanpwaldon/pen/ExpZVQB

### tsParticles Version

2.7.1

### Which library are you using?

Vanilla (tsparticles)

### tsParticles Configuration

```ts
tsParticles.load('particles', {
  fpsLimit: 120,
  fullScreen: false,
  detectRetina: true,
  background: {
    image: 'url(https://images.unsplash.com/photo-1468657988500-aca2be09f4c6)',
    position: '50% 50%',
    size: 'cover',
  },
  particles: {
    color: {
      value: '#fff',
    },
    move: {
      speed: 1,
      enable: true,
      random: false,
      outMode: 'out',
      straight: true,
      direction: 'bottom',
      trail: {
        enable: true,
        length: 20,
        fillColor: {
          value: '#fff',
        },
      },
    },
    shape: {
      type: 'circle',
    },
    size: {
      value: 1,
    },
  },
})
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by matteobruni** [comment]
Duplicate of #4576, #3495, #3078, #4870, #2446

**Comment by ryanpwaldon** [comment]
Thanks @matteobruni. Could we add a `fillImage` property to `trail` or is the performance impact an issue?

**Comment by matteobruni** [comment]
Performance not, it's drawing a picture once per frame, not a big deal.

This is what it could look like: https://codepen.io/matteobruni/pen/PoBWOpb

I'm unsure about the final result and I'd like to know your opinion about that. If it's what you thought, it's something that could be added easily to the project.

**Comment by ryanpwaldon** [comment]
Thanks @matteobruni! This is great!

For context – I'd like to overlay particles (with the trail effect) over a map to provide weather insights.

It's a shame the trail imprints are so visible with the image backdrop.

I made a tweak that removes the imprints – https://codepen.io/ryanpwaldon/pen/VwBbvxo (see lines 80 - 82) – but it only works if all particles are of the same color (this works great for my use case, but I'm not sure whether it's something you would want to add into the library as well?).

Are there any other possibilities for removing these imprints, or is drawing each trail individually the only known solution?

**Comment by matteobruni** [comment]
> Thanks @matteobruni! This is great!
> 
> For context – I'd like to overlay particles (with the trail effect) over a map to provide weather insights.
> 
> It's a shame the trail imprints are so visible with the image backdrop.
> 
> I made a tweak that removes the imprints – https://codepen.io/ryanpwaldon/pen/VwBbvxo (see lines 80 - 82) – but it only works if all particles are of the same color (this works great for my use case, but I'm not sure whether it's something you would want to add into the library as well?).
> 
> Are there any other possibilities for removing these imprints, or is drawing each trail individually the only known solution?

The imprints are an approximation issue that all browsers have, since they have some rounded values that makes the opacity overlay fail to reach `1` correctly. For the colors it's something configurable, so if the final user is fine with that color it's not a problem for me. Glad to hear that is what you want, it will be included in v3 (PR #4874)

**Comment by ryanpwaldon** [comment]
Thanks so much Matteo!

**Comment by matteobruni** [comment]
Released `2.8.0` with new `fill` section.

Instead of `fillColor`, the new property is like this: 

```
fill: {
    color: "#abcdef"
}
```

or

```
fill: {
    image: "http://foo.bar/baz.png"
}
```
